v2/tests/log_context_property_type_ascii_char_ptr_ut/log_context_property_type_ascii_char_ptr_ut.c

```diff
@@ -0,0 +1,530 @@
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_017: [ If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+static void ascii_char_ptr_to_string_with_NULL_buffer_and_non_zero_buffer_length_fails(void)
+{
+    // arrange
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"cucu", NULL, 1);
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_002: [ If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall return the length of the string pointed to by `property_value`. ]*/
+static void ascii_char_ptr_to_string_with_NULL_buffer_and_zero_buffer_length_returns_string_length(void)
+{
+    // arrange
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"cucu", NULL, 0);
+
+    // assert
+    POOR_MANS_ASSERT(result == 4);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_003: [ Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall copy the string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%s` and pass in the values list the `const char*` value pointed to be `property_value`. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_004: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall succeed and return the result of `snprintf`. ]*/
+static void ascii_char_ptr_to_string_copies_the_string(void)
+{
+    // arrange
+    char buffer[4];
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"bau", buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result == 3);
+    POOR_MANS_ASSERT(strcmp(buffer, "bau") == 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_005: [ If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+static void when_snprintf_fails_ascii_char_ptr_to_string_also_fails(void)
+{
+    // arrange
+    char buffer[4];
```

Comment:
>buffer[4]; [](http://example.com/codeflow?start=9&length=10)

What happens if buffer's size is 2?

Other data types have truncation... as a test.

Reply:
Truncation test added